File: app/src/main/java/com/kickstarter/mock/factories/PushNotificationEnvelopeFactory.java
package com.kickstarter.mock.factories;

import com.kickstarter.models.pushdata.GCM;
import com.kickstarter.services.apiresponses.PushNotificationEnvelope;

import androidx.annotation.NonNull;

public final class PushNotificationEnvelopeFactory {
  private PushNotificationEnvelopeFactory() {}

  public static @NonNull PushNotificationEnvelope envelope() {
    final GCM gcm = GCM.builder()
      .alert("You've received a new push notification")
      .title("Hello")
      .build();

    return PushNotificationEnvelope.builder()
      .gcm(gcm)
      .build();
  }
}

File: app/src/main/java/com/kickstarter/mock/factories/PushNotificationEnvelopeFactory.kt
package com.kickstarter.mock.factories

import com.kickstarter.models.pushdata.GCM.Companion.builder
import com.kickstarter.services.apiresponses.PushNotificationEnvelope

object PushNotificationEnvelopeFactory {
    fun envelope(): PushNotificationEnvelope {
        val gcm = builder()
            .alert("You've received a new push notification")
            .title("Hello")
            .build()
        return PushNotificationEnvelope.builder()
            .gcm(gcm)
            .build()
    }
}
